src/Mvc/Mvc.Core/test/ModelBinding/Validation/DefaultObjectValidatorTests.cs

```diff
@@ -1305,26 +1306,32 @@ public void Validate_WorksIfObjectGraphIsSmallerThanMaxDepth()
         }
 
         [Fact]
-        public void Validate_Throws_WithMaxDepth_1()
+        public void Validate_DoesNotThrow_IfNumberOfErrorsAfterReachingMaxAllowedErrors_GoesOverMaxDepth()
         {
             // Arrange
-            var maxDepth = 1;
-            var expected = $"ValidationVisitor exceeded the maximum configured validation depth '{maxDepth}' when validating property '{nameof(DepthObject.Depth)}' on type '{typeof(DepthObject)}'. " +
-                "This may indicate a very deep or infinitely recursive object graph. Consider modifying 'MvcOptions.MaxValidationDepth' or suppressing validation on the model type.";
+            var maxDepth = 4;
             _options.MaxValidationDepth = maxDepth;
             var actionContext = new ActionContext();
+            actionContext.ModelState.MaxAllowedErrors = 2;
             var validator = CreateValidator();
-            var model = new DepthObject(maxDepth + 1);
+            var model = new List<ModelWithRequiredProperty>
+            {
+                new ModelWithRequiredProperty(), new ModelWithRequiredProperty(),
+                // After the first 2 items we will reach MaxAllowedErrors
+                // If we add items without popping after having reached max validation,
+                // with 4 more items (on top of the list) we would go over max depth of 4
+                new ModelWithRequiredProperty(), new ModelWithRequiredProperty(),
+                new ModelWithRequiredProperty(), new ModelWithRequiredProperty(),
+            };
+
             var validationState = new ValidationStateDictionary
             {
                 { model, new ValidationStateEntry() }
             };
-            var method = GetType().GetMethod(nameof(Validate_Throws_ForTopLevelMetadataData), BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
This felt like something added by accident.

Reply:
yeah seems like good cleanup 👍